app/assets/javascripts/models/data/preferences.js.coffee

```diff
@@ -24,11 +25,16 @@ ns.Preferences = do (ko
         @fromJson(data)
         @isLoaded(true)
       else
-        getJSON '/users/site_preferences', (data, status, xhr) =>
-          console.log "Loaded site preferences, #{JSON.stringify(data)}"
-
-          @fromJson(data)
-          @isLoaded(true)
+        $.ajax '/users/site_preferences',
+          type: 'GET'
+          dataType: 'json',
+          async: false,
```

Comment:
@mightynimble - this is where I got the idea for using setTimeout to check preferences for the initial modal.